SDWebImage/UIImage+GIF.m

```diff
@@ -87,7 +87,9 @@ + (UIImage *)sd_animatedGIFNamed:(NSString *)name {
     CGFloat scale = [UIScreen mainScreen].scale;
 
     if (scale > 1.0f) {
-        NSString *retinaPath = [[NSBundle mainBundle] pathForResource:[name stringByAppendingString:@"@2x"] ofType:@"gif"];
+        int imgScale = [NSNumber numberWithFloat:scale].intValue;
+        NSString *scaleFactor = [NSString stringWithFormat:@"@%ix",imgScale];
```

Comment:
then the string format will use `%lu` (see https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/Strings/Articles/formatSpecifiers.html for NSUInteger)